test/Factories.py

```diff
@@ -365,3 +365,74 @@ def writeOutGradescopeGrades(_fileName: str, _gradeSheet: list[str]):
             gradesheetWriter.write(f"{line}\n")
 
         gradesheetWriter.close()
+
+    @staticmethod
+    def generateStudentRunestoneGradesForAssignment(_studentRoster: list[dict], _assignmentName: str,
+                                                     _assignmentPoints: int):
+        """
+        :Description:
+
+        This function generates a gradescope gradesheet for the student roster. It will randomly generate scores, missing assignments, and lateness.
+
+        Calls ``Factories.generateSingleStudentGradescopeGrade`` internally.
+        Calls ``Factories.writeOutGradescopeGrades`` internally.
+
+        This data will be written out to  ``./gradescope/{_assignmentName}_test_scores.csv``.
+
+        :param _studentRoster: A student roster generated by ``Factories.generateStudentRoster``.
+        :param _assignmentName: The name of the assignment that grades are being generated for.
+        :param _assignmentPoints: The max number of points for this assignment
+        """
+
+        # seed random for reproducibility
+        random.seed(12345)
+
+        # On average students get an 75% - 80% with a standard deviation of 12% - 15%.
+        #  Using a 75% avg and a 15% sd for this
+        gradeDist: stats.truncnorm = generateTruncNorm(mean=_assignmentPoints * .75, sd=_assignmentPoints * .15,
+                                                       lower=0, upper=_assignmentPoints)
+
+        # Seed scipy for reproducibility
+        numpyGenerator = numpy_rand.Generator(numpy_rand.PCG64(12345))
+        gradeDist.random_state = numpyGenerator
+        gradeSheet = []
+
+        # three lines that are in the runestone format parser
+        gradeSheet.append("0, 0")
+        gradeSheet.append("0, 3") # total points
+        gradeSheet.append("0, 0")
+        for student in _studentRoster:
+            studentScore = gradeDist.rvs()
+            gradeSheet.append(f"{student['email']},{round(studentScore,2) * 10}%")
```

Comment:
Is there a lateness in runestone? If so we should be generating it.

Reply:
Not that I can tell after investigation on our current admin panel, but I'll reach out to the Runestone people.